hal-x86_64/src/interrupt/mod.rs

```diff
@@ -57,21 +57,15 @@ impl Handlers<crate::X64> for TestHandlersImpl {
     where
         C: ctx::Context<Arch = crate::X64> + ctx::PageFault,
     {
-        let mut vga = vga::writer();
-        vga.set_color(vga::ColorSpec::new(vga::Color::Red, vga::Color::Black));
-        writeln!(&mut vga, "page fault\n{:#?}", cx.registers()).unwrap();
-        vga.set_color(vga::ColorSpec::new(vga::Color::Green, vga::Color::Black));
+        tracing::error!(registers = ?cx.registers(), "page fault");
     }
 
     #[inline(never)]
     fn code_fault<C>(cx: C)
     where
         C: ctx::Context<Arch = crate::X64> + ctx::CodeFault,
     {
-        let mut vga = vga::writer();
-        vga.set_color(vga::ColorSpec::new(vga::Color::Red, vga::Color::Black));
-        writeln!(&mut vga, "code fault\n{:#?}", cx.registers()).unwrap();
-        vga.set_color(vga::ColorSpec::new(vga::Color::Green, vga::Color::Black));
+        tracing::error!(registers = ?cx.registers(), "code fault");
         loop {}
```

Comment:
I suppose we could probably `panic!` here and in the other fatal interrupts as well, rather than `loop {}`-ing.

Reply:
in a follow-up, i am planning on making them call into a "kernel oops" handler that the panic handler also calls into, but with different args. in the future, some of these will at least occasionally be recoverable (in user code)